### PR TITLE
feat: DTOSS 5205 diagnostic settings module fahad

### PR DESCRIPTION
**Pull request by fahadbaigNC**
## Description

Changes made to retrieve output from audit tfstatefile to consume the log analytics workspace ID

## Type of changes

- [x] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: infrastructure/tf-core/locals.tf
Comment (mrlockstar) [inline infrastructure/tf-core/locals.tf:1]: Can you try this instead please :) !  

primary_region = [for k, v in var.regions : k if v.is_primary_region][0]

File: infrastructure/tf-audit/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# output "log_analytics_workspace_id" {
```
Comment (mrlockstar) [inline infrastructure/tf-audit/outputs.tf:1]: Delete these commented out blocks.

File: infrastructure/tf-core/locals.tf
Comment (fahadbaigNC) [reply infrastructure/tf-core/locals.tf:1]: Will put off refactoring for later.

File: infrastructure/tf-audit/outputs.tf
Comment (fahadbaigNC) [inline infrastructure/tf-audit/outputs.tf:1]: @mrlockstar removed comments. Thanks for that

**Review by rfk-nc — DISMISSED** [review]

**Review by MacMur85 — APPROVED** [review]
Approved. Code changes tested and applied: 

https://dev.azure.com/nhse-dtos/dtos-cohort-manager/_build/results?buildId=5070&view=logs&j=c970d263-77bb-52d7-1861-5b2637d7126d&t=d474bb48-30f3-5027-0539-b68c9f63a53a

**Review by josielsouzanordcloud — APPROVED** [review]

**Pull request merged**